src/data/finset/basic.lean

```diff
@@ -11,19 +11,124 @@ import tactic.nth_rewrite
 /-!
 # Finite sets
 
-mathlib has several different models for finite sets,
-and it can be confusing when you're first getting used to them!
+Terms of type `finset α` are one way of talking about finite subsets of `α` in mathlib.
+Below, `finset α` is defined as a structure with 2 fields:
+
+  1. `val` is a `multiset α` of elements;
+  2. `nodup` is a proof that `val` has no duplicates.
+
+Finsets in Lean are constructive in that they have an underlying `list` that enumerates their
+elements. In particular, any function that uses the data of the underlying list cannot depend on its
+ordering. This is handled on the `multiset` level by multiset API, so in most cases one needn't
+worry about it explicitly.
+
+Finsets give a basic foundation for defining finite sums and products over types:
+
+  1. `∑ i in (s : finset α), f i`;
+  2. `∏ i in (s : finset α), f i`.
+
+Lean refers to these operations as `big_operator`s.
+More information can be found in `algebra.big_operators.basic`.
+
+Finsets are directly used to define fintypes in Lean.
+A `fintype α` instance for a type `α` consists of
+a universal `finset α` containing every term of `α`, called `univ`. See `data.fintype.basic`.
+There is also `univ'`, the noncomputable partner to `univ`,
+which is defined to be `α` as a finset if `α` is finite,
+and the empty finset otherwise. See `data.finsum`.
```

Comment:
Why is there a reference to `data.finsum` here?

Reply:
Oops. Copy/paste error, probably.